src/object/srv_obj.c

```diff
@@ -1160,6 +1175,15 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	}
 
 	if (obj_rpc_is_update(rpc)) {
+		if (daos_csummer_get_dedup(cont_hdl->sch_csummer) &&
+		    daos_csummer_get_dedupverify(cont_hdl->sch_csummer)) {
+			/**
+			 * XXX Need to check for dedup block here and memcmp
+			 * if memcmp fails, just go on with the original buffer,
+			 * otherwise, just drop it
+			 */
```

Comment:
@NiuYawei I would appreciate your help to finish the implementation of the memcmp flavor here.
I changed the code above in this function to use the regular patch (non-dedup) when the verify/memcmp option is selected, but i am not sure how to play with the different bio_vec here. Thanks in advance (feel free to direction updated this patch).

Reply:
Sure, I'll update the patch.

Reply:
@johannlombardi dedup verify is implemented in the 2nd commit of this PR, please take a look. Thanks.